### PR TITLE
Ability to disable Groovy version check

### PR DESCRIPTION
**Pull request by szpak**
In production Spock version. For early adopters of the further Groovy
versions (such as 4.x). And to use in the Groovy 4.x build itself - instead of the snapshot version.

Before Spock 2.0-M1 people were able to use only 1.3-SNAPSHOT with Groovy 3, even though, in the majority of cases, it just worked.

If you have a good idea how to test it in a better way, please let me know.

---
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/1164)

### CODE REVIEW

File: spock-core/src/main/java/org/spockframework/util/VersionChecker.java [hunk 1]
```diff
@@ -15,15 +15,27 @@
...
+  //visibility for testing
+  static final String DISABLE_GROOVY_VERSION_CHECK_PROPERTY_NAME = "spock.iKnowWhatImDoing.disableGroovyVersionCheck";
 
-  public static void checkGroovyVersion(String whoIsChecking) {
-    if (!compatibleGroovyVersion) throw new IncompatibleGroovyVersionException(String.format(
-"The Spock %s cannot execute because Spock %s is not compatible with Groovy %s. For more information, see http://docs.spockframework.org\n" +
+  private static final boolean compatibleGroovyVersion = SpockReleaseInfo.isCompatibleGroovyVersion(GroovyReleaseInfo.getVersion());
+
+  public void checkGroovyVersion(String whoIsChecking) {
+    if (!isCompatibleGroovyVersion() && !isVersionCheckDisabled()) throw new IncompatibleGroovyVersionException(String.format(
```
Comment (leonard84) [inline spock-core/src/main/java/org/spockframework/util/VersionChecker.java:24]: I know that we don't log often but I think that we could log to std.err that the check failed but was ignored due to the property, WDYT?

File: docs/release_notes.adoc [hunk 2]
```diff
@@ -207,6 +207,8 @@ Spy(constructorArgs: [null as String, (Pattern) null])
 def "I'll run everywhere but on Windows"() { ... }
 ----
 
+- Execution with the unsupported Groovy version can enforced with `-Dspock.iKnowWhatImDoing.disableGroovyVersionCheck=true`
```
Comment (leonard84) [inline docs/release_notes.adoc:210]: > Execution with the unsupported Groovy version can enforced with

I would rephrase that to

`Execution with an unsupported Groovy version can allowed with`

File: spock-core/src/main/java/org/spockframework/util/VersionChecker.java (hunk 1 above)
Comment (szpak) [reply spock-core/src/main/java/org/spockframework/util/VersionChecker.java]: This flag shouldn't be used on a regular basis, so cluttering the stderr to keep users aware sounds fine for me. Changed.

We could check the stderr in tests to verify that message, but 2 ways I know use JUnit 4 rules which we removed from spock-spec ;).

File: docs/release_notes.adoc (hunk 2 above)
Comment (szpak) [reply docs/release_notes.adoc]: Chagned.
